setup.py

```diff
@@ -14,6 +14,7 @@
         "sphinx-issues==1.2.0",
         "alabaster==0.7.12",
         "sphinx-version-warning==1.1.2",
+        "autodocsumm==0.1.13",
     ],
```

Comment:
Do we need to add a dependency to https://github.com/Chilipp/autodocsumm ?

What do we need it for that can't be achieved with sphinx autosummary (https://www.sphinx-doc.org/en/master/usage/extensions/autosummary.html)?

Reply:
With `sphinx.ext.autosummary`, you need to enumerate every item that you want to include in the documentation (until Sphinx 3.1 is released, where the `recursive` parameter is introduced). This means any new methods, classes and attributes need to be added/removed manually in the documentation as well.

`autodocsumm` provides a way to quickly add a summary table at the top of any [autodoc](https://www.sphinx-doc.org/en/master/usage/extensions/autodoc.html) directive. Seeing as there are already docs-building requirements anyway (sphinx etc), another small dependency won't affect CI or the development workflow.

If you don't want the summary table added, it's easy for me to revert the addition of the summary table.